### ISSUE TITLE
When the Authorization header is set, but the value is empty an exception is thrown

### ISSUE THREAD

**Issue by VirajSalaka** [wso2/product-microgateway#1098]
### Description:
When the Authorization header is set, but the value is empty an exception is thrown.
```
2020-03-03 14:57:50 INFO  ServerLogReader:114 - error: {ballerina/mime}HeaderNotFound message=Http header does not exist
2020-03-03 14:57:50 INFO  ServerLogReader:114 -         at ballerina.mime:externGetHeader(natives.bal:534)
2020-03-03 14:57:50 INFO  ServerLogReader:114 -            ballerina.mime.Entity:getHeader(natives.bal:389)
2020-03-03 14:57:50 INFO  ServerLogReader:114 -            ballerina.http.Request:getHeader(http_request.bal:132)
2020-03-03 14:57:50 INFO  ServerLogReader:114 -            wso2.gateway.src.gateway.filters.pre_authn_filter:doAuthnFilterRequest(/src/gateway/filters/pre_authn_filter.bal:90)
2020-03-03 14:57:50 INFO  ServerLogReader:114 -            wso2.gateway.PreAuthnFilter:filterRequest(/src/gateway/filters/pre_authn_filter.bal:36)
2020-03-03 14:57:50 INFO  ServerLogReader:114 -            wso2.GrpcProject__service_0:post10c85995bdf54defb7c9cec9f9fbe7d1(JwtAuthTestService__1_0_0.bal:95)
```
Detailed description is included in the following issue.
https://github.com/ballerina-platform/ballerina-lang/issues/21441

Until this is fixed, we can introduce a workaround to trap the getHeaders function and proceed. Once the issue is fixed in ballerina, that workaround won't be needed.

### Steps to reproduce:
Please refer to https://github.com/ballerina-platform/ballerina-lang/issues/21441

### Affected Product Version:
3.1.0-beta

### Environment details (with versions):
- OS: MacOS Mojave
- Client:
- Env (Docker/K8s):

---
#### Suggested Labels:
Affected/3.1.0-beta

**Comment by VirajSalaka** [comment]
The expected workaround is to trap the getHeader function, but still it won't be the proper solution since getHeader method is called in multiple occasions. Since it is a bug in ballerina code, it would be better to wait till the patch is released so that the gateway code does not need to change.

**Comment by Rajith90** [comment]
We will have to update to ballerina 1.1.5 version to get the fix

**Comment by VirajSalaka** [comment]
Resolved in 3.2.0-alpha as the ballerina version is upgraded. Hence closing the issue.
